src/utils/retry.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Executes a Promise-based function multiple times with retries.
```

Comment:
Promise-based isn't a thing - call it an async fn.

```suggestion
 * Executes an async function multiple times with retries.
```